3-enrich/stream-enrich/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sinks/KinesisSink.scala

```diff
@@ -55,12 +50,12 @@ object KinesisSink {
       .withCredentials(provider)
       .withEndpointConfiguration(endpointConfiguration)
       .build()
-    _ <- streamExists(client, streamName).leftMap(_.getMessage)
+    exists <- streamExists(client, streamName).leftMap(_.getMessage)
       .flatMap { b =>
```

Comment:
This `flatMap` can be changed to `streamExists(client, streamName).ensure("Kinesis stream $streamName doesn't exist")(_ == true)`